### PR TITLE
Add tut contrib module

### PR DESCRIPTION
**Pull request by DavidGregory084**
- [x] Add tut module
- [x] Add tests
- [x] Add docs

### CODE REVIEW

File: contrib/tut/src/mill/contrib/tut/TutModule.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  def tutClasspath: T[Agg[PathRef]] = tutJar() ++ compileClasspath()
+  def tutScalacPluginIvyDeps: T[Agg[Dep]] = scalacPluginIvyDeps()
+  def tutNameFilter: T[Regex] = T { """.*\.(md|markdown|txt|htm|html)""".r }
+  def tutScalacOptions: T[Seq[String]] = scalacOptions()
+  def tutVersion: T[String] = "0.6.7"
+
+  def tutJar: T[Agg[PathRef]] = T {
+    Lib.resolveDependencies(
+      repositories :+ MavenRepository(s"https://dl.bintray.com/tpolecat/maven"),
+      Lib.depToDependency(_, "2.12.4"),
```
Comment (DavidGregory084) [inline contrib/tut/src/mill/contrib/tut/TutModule.scala]: Is there a "millScalaVersion" somewhere that we can use for definitions like this?

Comment (DavidGregory084) [reply contrib/tut/src/mill/contrib/tut/TutModule.scala]: Actually I don't think that makes sense here - we want Tut to run with the `scalaVersion` of the module

**Comment by DavidGregory084** [comment]
I want to add the `localClasspath` to the `tutClasspath` but this causes the same issue that @nh13 experience on their coverage PR

I'm trying to figure out what causes that problem

**Comment by DavidGregory084** [comment]
It seems like this is due to the system property MILL_VERSION being unset - this is passed directly to coursier as a version for mill-scalalib-worker

**Comment by nh13** [comment]
@DavidGregory084 I'll definitely keep an eye on this.  I had issues with ivy dependency resolution too in the latest version of my PR :/.

**Comment by DavidGregory084** [comment]
@nh13 I had to set the testArgs in the module in order to avert the issue

**Comment by DavidGregory084** [comment]
I'm still running into some funny business with the classpath not being used by the scala interpreter. I think there might be some SBT specific stuff in there somewhere, maybe this: https://github.com/tpolecat/tut/blob/series/0.6.x/modules/core/src/main/scala/tut/FileIO.scala#L79

I'll keep chipping away in any case

File: build.sc [hunk 2]
```diff
@@ -250,6 +250,10 @@ object contrib extends MillModule {
     }
    }
 
+  object tut extends MillModule {
+    def moduleDeps = Seq(scalalib)
+    def testArgs = Seq("-DMILL_VERSION=" + build.publishVersion()._1)
```
Comment (nh13) [inline build.sc]: should this be `_.2` like it is in other parts of the build file?

Comment (DavidGregory084) [reply build.sc]: I tried that, but unfortunately in a project with local changes it just meant that Coursier was unable to resolve the dependency (as MILL_VERSION is some unpublished version of mill)

**Comment by DavidGregory084** [comment]
Ok, I think this is ready to go now

I was disappointed to have to shell out to call Tut but there was no obvious way to change it to make sure it used the class path of the context class loader in the interpreter (I may come back to that later).

I also added a test using libraries and scalac plugins to make sure that worked

**Comment by lefou** [comment]
I'd love to see the already good task documentation also as ScalaDoc. After all, mill is scala source based and at least when used in IntelliJ, the ScalaDoc would be immediately visible when configuring the module.

**Comment by DavidGregory084** [comment]
Great idea!

**Comment by lefou** [comment]
LGTM.

@DavidGregory084 Do you plan to add the tasks ScalaDoc? As soon as you give green light, I can merge this PR.

**Comment by DavidGregory084** [comment]
@lefou I should be able to take a look at that this evening, or you can merge this one now and I'll send a follow up PR to do that for all the contrib modules? I don't mind which :)

**Comment by lefou** [comment]
@DavidGregory084 As this PR needs a conflict resolve, I'd rather wait for your rebase and ScalaDoc commit than merge as is.

**Comment by DavidGregory084** [comment]
Okay @lefou I think this is ready to go pending your review

File: contrib/tut/src/mill/contrib/tut/TutModule.scala [hunk 3]
```diff
@@ -6,23 +6,96 @@ import coursier.MavenRepository
...
+      "-Ywarn-unused:imports",
+      "-Ywarn-unused-import",
+      "-Ywarn-dead-code",
+      "-Xfatal-warnings"
+    ))
+
+  /**
+    * The version of Tut to use.
+    */
   def tutVersion: T[String] = "0.6.7"
```
Comment (lefou) [inline contrib/tut/src/mill/contrib/tut/TutModule.scala]: To follow the same pattern as in other mill modules (e.g. ScalaModule), we should not hard code any versions. Hence this field should be mandatory.

Comment (DavidGregory084) [reply contrib/tut/src/mill/contrib/tut/TutModule.scala]: Yep, seems reasonable

Comment (DavidGregory084) [reply contrib/tut/src/mill/contrib/tut/TutModule.scala]: OK, the latest commit removes the default version here

**Comment by lefou** [comment]
Thank you!

**Pull request merged**